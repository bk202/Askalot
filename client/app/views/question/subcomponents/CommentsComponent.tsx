import * as React from "react";
import {CommentDto} from "../../../../../server/dtos/q&a/CommentDto";
import PropTypes from 'prop-types';
import {withStyles, createStyleSheet} from 'material-ui/styles';
import List, {ListItem, ListItemText} from 'material-ui/List';
import Divider from 'material-ui/Divider';
import Icon from 'material-ui/Icon';
import {CSSProperties} from "react";
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import IconButton from 'material-ui/IconButton'
import {UserDto} from "../../../../../server/dtos/auth/UserDto";

export interface CommentsComponentProps {
    comments: CommentDto[];
    user : UserDto;
    onCommentsSubmit : (comments: CommentDto[]) => void;
}

export interface CommentsComponentState {
    inputMode: boolean;
    commentContent: string;
    errorMsg: string;
}

const styleSheet: CSSProperties = {
    root: {
        width: '100%',
        maxWidth: '360px'
    }
};

export class CommentsComponent extends React.Component<CommentsComponentProps, CommentsComponentState> {
    constructor(props) {
        super (props);
        this.state = {
            inputMode: false,
            commentContent: "",
            errorMsg: "",
        };
    }


    addNewComment = () => {
        if(this.state.commentContent){
            this.setState({errorMsg: ""});

        }
        else{
            this.setState({errorMsg: "Cannot submit empty comment."});
        }
    }

    onCommentChange = (event)  => {
        this.setState({commentContent: event.target.value});
    }

    onErrMsg = () => {
        if(this.state.errorMsg != ""){
            return(
                <p><mark>{this.state.errorMsg}</mark></p>
            )
        }
    }

    renderInputCommentBox() {
        if (this.state.inputMode) {
            return (
                <div>
                    {this.onErrMsg()}
                    <form>
                        <textarea rows={4} cols={50} value={this.state.commentContent} onChange={this.onCommentChange}></textarea>
                    </form>
                    <Button raised color="primary" className={styleSheet.root} onClick={this.addNewComment}>
                        Submit
                    </Button>
                </div>
            )
        }
    }

    render() {
        return (
            <div>
                <List className={"Comments"}>
                    <ListItem button>
                        <ListItemText primary="Inbox"/>
                    </ListItem>
                    <Divider light/>
                    <ListItem button>
                        <ListItemText primary="Drafts"/>
                    </ListItem>
                    <Divider />
                    <ListItem button>
                        <ListItemText primary="Trash"/>
                    </ListItem>
                </List>
                <div className={styleSheet.root}>
                    <IconButton>
                    <Icon onClick={() => this.setState({inputMode: !this.state.inputMode}) }>add_circle</Icon>
                    </IconButton>
                </div>
                {this.renderInputCommentBox()}
            </div>
        );
    }
}