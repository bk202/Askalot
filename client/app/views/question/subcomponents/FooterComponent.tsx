import * as React from "react";
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import {ThumbComponent} from "../../../components/Forms/ThumbComponent";

interface props {
    onUpVote: () => void;
    onDownVote: () => void;
    upVotes: number;
    downVotes: number;
    author: UserDto;
    createdUtc: Date;
}

export class FooterComponent extends React.Component<props> {


    render() {
        let question = {
            upVotes: 0
        };
        return (
            <div>
                <ThumbComponent
                    value={this.props.upVotes}
                    thumbUp={true}
                    onClick={this.props.onUpVote} />
                <ThumbComponent
                    value={this.props.downVotes}
                    thumbUp={false}
                    onClick={this.props.onDownVote} />
                <div style={{color: "grey", fontSize: 10, textAlign: "right"}}>
                    {this.props.createdUtc && <div>Posted on {this.props.createdUtc}</div>}
                    <br/>
                    by {this.props.author.username}
                </div>
            </div>
        );
    }
}