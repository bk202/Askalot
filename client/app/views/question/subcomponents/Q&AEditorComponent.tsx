import * as React from "react";
import {Component} from "react";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import {EditorState} from "draft-js";
import {CustomEditor} from "../../../components/CustomEditor/CustomEditor";
import Button from "material-ui/Button";
import QuestionPreview = FrontEndQuestionModels.QuestionPreview;

export interface QAEditorProps{
    value: EditorState;
    onChange: (state: EditorState) => any;
    onSubmit: () => any;
    readOnly?: boolean;
    style?:any;
}

export class QAEditorComponent extends Component<QAEditorProps> {
    render() {
        const readOnly = this.props.readOnly;
        return (
            <div style={{marginBottom: 15}}>
                <CustomEditor value={this.props.value} onChange={this.props.onChange} readOnly={readOnly}
                            border={!readOnly} style={this.props.style}
                />
                {
                    !this.props.readOnly &&
                    <div style={{textAlign: "right"}}>
                        <Button onClick={this.props.onSubmit}>save</Button>
                    </div>
                }
            </div>
        )
    }
}