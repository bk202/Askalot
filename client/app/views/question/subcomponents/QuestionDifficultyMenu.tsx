import * as React from "react";
import {DropDownSelect} from "../../../components/DropDownSelect";
import {DifficultyLevel, QuestionEducationLevel} from "../../../../../server/enums/QuestionEducationLevel";
import {getDropDownDataFromNumericalEnum, getDropDownDataFromStringEnum} from "../../../utils/utils";
import {QuestionDifficulty} from "../../../../../server/models/Question";

interface props {
    onDifficultyChange: (diff: QuestionDifficulty) => void;
    difficulty: QuestionDifficulty;
}

export class QuestionDifficultyMenu extends React.Component<props> {
    render() {
        return (
            <div>
                <DropDownSelect
                    placeholder="Question Level"
                    data={getDropDownDataFromStringEnum(QuestionEducationLevel)}
                    onChange={(educationLevel) => this.props.onDifficultyChange({
                        educationLevel, difficultyLevel: this.props.difficulty.difficultyLevel
                    })}
                    defaultValue={QuestionEducationLevel.NOT_SPECIFIED}
                />
                { this.props.difficulty.educationLevel != QuestionEducationLevel.NOT_SPECIFIED &&
                <DropDownSelect
                    placeholder="Difficulty Level"
                    data={getDropDownDataFromNumericalEnum(DifficultyLevel)}
                    onChange={(difficultyLevel) => this.props.onDifficultyChange({
                        educationLevel: this.props.difficulty.educationLevel, difficultyLevel
                    })}
                    defaultValue={DifficultyLevel.NOT_SPECIFIED}
                />}
            </div>
        );
    }
}