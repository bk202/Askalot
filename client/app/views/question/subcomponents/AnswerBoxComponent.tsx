import * as React from "react";
import {Component} from "react";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import Divider from "material-ui/Divider";
import {QAEditorComponent} from "./Q&AEditorComponent";
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import Button from "material-ui/Button";
import Paper from "material-ui/Paper";
import Grid from "material-ui/Grid";
import QuestionPreview = FrontEndQuestionModels.QuestionPreview;
import Question = FrontEndQuestionModels.Question;
import Answer = FrontEndQuestionModels.Answer;

export interface AnswerBoxComponentProps {
    onAnswerChange: (answer: Answer) => void;
    onSubmit: () => void;
    onEditClick: () => void;
    user: UserDto;
    answer: Answer;
    editMode: boolean;
    resetAnswer?: () => void;
}

const paperStyle = {height: "100%", padding: 15, paddingBottom: 0};

export class AnswerBoxComponent extends Component<AnswerBoxComponentProps> {

    onContentChange = (editorState) => {
        let question = this.props.answer;
        question.content = editorState;
        this.props.onAnswerChange(question);
    };

    render() {
        const answer = {...this.props.answer};
        const editable = (this.props.user && this.props.user.username === answer.author.username);
        return (
            <Paper style={paperStyle}>
                <div>
                    <Grid container justify="flex-end">
                        <Grid item>
                            {editable && !this.props.editMode && <Button onClick={this.props.onEditClick}>Edit</Button>}
                        </Grid>
                    </Grid>
                    <QAEditorComponent value={this.props.answer.content} onChange={this.onContentChange}
                                       onSubmit={this.props.onSubmit} readOnly={!this.props.editMode}
                                       style={{fontSize: 14}} reset={this.props.resetAnswer}
                    />
                    <Divider/>
                    <p style={{color: "grey", fontSize: 10, textAlign: "right"}}>
                        {answer.createdUtc && <div>Posted on {answer.createdUtc}</div>}
                        <br/>by {answer.author.username}
                    </p>
                </div>
            </Paper>
        )
    }
}