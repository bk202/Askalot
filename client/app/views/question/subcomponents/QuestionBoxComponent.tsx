import * as React from "react";
import {Component} from "react";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import {EditableMultiPurposeHeader} from "../../../components/Headers/EditableMultiPurposeHeader";
import {ChipListComponent} from "../../../components/ChipListComponent";
import Divider from "material-ui/Divider";
import {QAEditorComponent} from "./Q&AEditorComponent";
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import Button from "material-ui/Button";
import Paper from "material-ui/Paper";
import QuestionPreview = FrontEndQuestionModels.QuestionPreview;
import Question = FrontEndQuestionModels.Question;
import cloneQuestion = FrontEndQuestionModels.cloneQuestion;

export interface QuestionBoxComponentProps {
    onQuestionChange: (question: Question) => void;
    onSubmit: () => void;
    onEditClick?: () => void;
    user: UserDto;
    question: Question;
    editMode: boolean;
    resetQuestion? : () => void ;
}

let paperStyle = {height: "100%", padding: 15, paddingBottom: 0};

export class QuestionBoxComponent extends Component<QuestionBoxComponentProps> {
    onTitleChange = (event) => {
        let question = cloneQuestion(this.props.question);
        question.title = event.target.value;
        this.props.onQuestionChange(question);
    };

    onContentChange = (editorState) => {
        let question = cloneQuestion(this.props.question);
        question.content = editorState;
        this.props.onQuestionChange(question);
    };

    render() {
        const question = {...this.props.question};
        const editable = this.props.user.username === question.author.username;
        let editButton;
        if (editable && !this.props.editMode){
            editButton = (
                <div style={{float: "right"}}>
                    <Button color="primary" onClick={this.props.onEditClick}>Edit</Button>
                </div>
            )
        }
        return (
            <Paper style={paperStyle}>
                <EditableMultiPurposeHeader value={question.title} editMode={this.props.editMode}
                                            onEditClick={this.props.onEditClick}
                                            onTitleChange={this.onTitleChange}/>
                {editButton}
                <Divider />
                <div>
                    <QAEditorComponent value={this.props.question.content} onChange={this.onContentChange}
                                       onSubmit={this.props.onSubmit} readOnly={!this.props.editMode}
                                        style={{fontSize: 14}} reset={this.props.resetQuestion}
                    />
                    <div>
                        <ChipListComponent chips={question.tags} keyName={"tag"}/>
                    </div>
                    <Divider/>
                    <p style={{color: "grey", fontSize: 10, textAlign: "right"}}>
                        Posted on {question.createdUtc}<br/>by {question.author.username}
                    </p>
                </div>
            </Paper>
        )
    }
}