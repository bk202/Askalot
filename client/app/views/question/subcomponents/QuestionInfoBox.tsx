import * as React from "react";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import Typography from "material-ui/Typography";
import {connect} from "react-redux";
import {AppStoreState} from "../../../stores/AppStore";
import Divider from "material-ui/Divider";
import {PublicityStatus} from "../../../../../server/enums/PublicityStatus";
import {convertEnumStringToViewString, getDropDownDataFromStringEnum} from "../../../utils/utils";
import {DifficultyLevel, QuestionEducationLevel} from "../../../../../server/enums/QuestionEducationLevel";
import {QuestionActions} from "../../../actions/QuestionActions";
import {QuestionDifficultyMenu} from "./QuestionDifficultyMenu";
import {QuestionDifficulty} from "../../../../../server/models/Question";
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import {DropDownSelect} from "../../../components/Forms/DropDownSelect";
import {QuestionEditorReducerState} from "../../../reducers/QuestionEditorReducer";
import Question = FrontEndQuestionModels.Question;
import cloneQuestion = FrontEndQuestionModels.cloneQuestion;
import {DisplayField} from "../../../components/Forms/DisplayField";
import {CategoryTypeEnum} from "../../../../../server/enums/CategoryTypeEnum";

interface stateToProps {
    question: Question;
    user: UserDto;
    questionEditorState: Question;
    edit: boolean;
}

interface DispatchProps {
    editQuestion: (question: Question) => void;
    changeQuestionEditorState: (state: QuestionEditorReducerState) => void;
}
interface props extends stateToProps, DispatchProps{}
export class QuestionInfoBox extends React.Component<props, {}> {

    onDifficultyChange = (difficulty: QuestionDifficulty) => {
        let question = cloneQuestion(this.props.questionEditorState);
        question.difficulty = difficulty;
        this.props.changeQuestionEditorState({
            edit: this.props.edit,
            question
        });
    };

    opnPublicityChange = (publicityStatus: PublicityStatus) => {
        let question = cloneQuestion(this.props.questionEditorState);
        question.publicityStatus = publicityStatus;
        this.props.changeQuestionEditorState({
            edit: this.props.edit,
            question
        });
    };

    onCategoryChange = (cat: CategoryTypeEnum) => {
        let question = cloneQuestion(this.props.questionEditorState);
        question.category = cat;
        this.props.changeQuestionEditorState({
            edit: this.props.edit,
            question
        });
    };

    row = (label: string, value: string) => {
        return <DisplayField label={label} value={value}/>
    };

    saveChanges = () => {
        this.props.editQuestion(this.props.questionEditorState);
    };

    view = () => {
        const question = this.props.question;
        return (
            <div>
                {this.row("Type", convertEnumStringToViewString(PublicityStatus[question.publicityStatus]))}
                {this.row("Category", convertEnumStringToViewString(CategoryTypeEnum[question.category]))}
                {this.row("Level",
                    convertEnumStringToViewString(QuestionEducationLevel[question.difficulty.educationLevel]))}
                {question.difficulty.educationLevel != QuestionEducationLevel.NOT_SPECIFIED &&
                this.row("Difficulty",
                    convertEnumStringToViewString(DifficultyLevel[question.difficulty.difficultyLevel]))}
            </div>
        )
    };
    editor = () => {
        return (
            <div>
                <DropDownSelect
                    placeholder="Publicity Status"
                    data={getDropDownDataFromStringEnum(PublicityStatus)}
                    value={this.props.questionEditorState.publicityStatus}
                    onChange={this.opnPublicityChange}
                />
                <DropDownSelect
                    placeholder="Category"
                    data={getDropDownDataFromStringEnum(CategoryTypeEnum)}
                    onChange={this.onCategoryChange}
                    value={this.props.questionEditorState.category}
                />
                <QuestionDifficultyMenu onDifficultyChange={this.onDifficultyChange}
                                        difficulty={this.props.questionEditorState.difficulty} />
            </div>
        );
    };

    render() {


        return (
            <div>
                <Divider />
                    {this.props.edit? this.editor(): this.view()}
                <Divider />
            </div>
        );
    }
}

const mapStateToProps = (state: AppStoreState) => ({
    question: state.questionPage.questionPage.question,
    user: state.auth.user,
    questionEditorState: state.questionEditorState.question,
    edit: state.questionEditorState.edit
});

const mapDispatchToProps = (dispatch): DispatchProps => ({
    editQuestion: (question: Question) => dispatch(QuestionActions.updateQuestion(question)),
    changeQuestionEditorState: (state: QuestionEditorReducerState) =>
        dispatch(QuestionActions.changeQuestionEditorState(state)),
});

export const QuestionInfoBoxView = connect<stateToProps, DispatchProps, any>(
    mapStateToProps,
    mapDispatchToProps
)(QuestionInfoBox);