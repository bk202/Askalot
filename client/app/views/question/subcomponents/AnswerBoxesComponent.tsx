import * as React from "react";
import {Component} from "react";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import {AnswerBoxComponent, AnswerBoxView} from "./AnswerBoxComponent";
import Button from "material-ui/Button";
import AddIcon from "material-ui-icons/Add";
import QuestionPreview = FrontEndQuestionModels.QuestionPreview;
import Question = FrontEndQuestionModels.Question;
import Answer = FrontEndQuestionModels.Answer;
import cloneAnswers = FrontEndQuestionModels.cloneAnswers;
import {connect} from "react-redux";
import {AppStoreState} from "../../../stores/AppStore";
import {AnswerActions} from "../../../actions/AnswerActions";
import {QuestionPageReducerState} from "../../../reducers/QuestionPageReducer";
import cloneAnswer = FrontEndQuestionModels.cloneAnswer;
import {Prompt} from "react-router";
export interface AnswerBoxesComponentProps {
}

interface props extends AnswerBoxesComponentProps, DispatchToProps, StateToProps {
}

export interface AnswerBoxesComponentState {
    answerId: string;
    editAnswer: boolean;
    answers: Answer[];
}

export class AnswerBoxesComponent extends Component<props, AnswerBoxesComponentState> {
    state = {answerId: undefined, editAnswer: false, answers: []};

    componentWillMount() {
        this.setState({answers: cloneAnswers(this.props.answers)});
    }

    componentWillReceiveProps(nextProps: props) {
        if (nextProps.answers != this.props.answers) {
            this.setState({editAnswer: false, answers: cloneAnswers(nextProps.answers)});
        }
    }

    componentWillUnmount() {
        this.resetAnswers();
    }

    onEditClick = (answer: Answer) => {
        this.setState({answerId: answer._id, editAnswer: true})
    };

    onAnswerChange = (answer: Answer) => {
        let answers = this.state.answers;
        answers = answers.map((ans) => {
            ans = cloneAnswer(ans);
            if (answer._id === ans._id) {
                return answer;
            }
            return ans;
        });
        this.setState({answers});
    };

    addAnswer = () => {
        let answers = cloneAnswers(this.state.answers);
        let answer = new Answer(this.props.question._id, this.props.user);
        answers.push(answer);
        this.setState({answerId: answer._id, editAnswer: true, answers});
    };

    resetAnswers = () => {
        this.setState({editAnswer: false, answerId: undefined});
        let answers = cloneAnswers(this.props.answers);
        this.setState({answers});
    };

    onSubmit = (answer: Answer) => {
        if (answer._id) {
            this.props.editAnswer(answer);
        } else {
            this.props.addAnswer(answer);
        }
    };

    render() {
        return (
            <div>
                <Prompt
                    when={this.state.editAnswer}
                    message={location => (
                        `All unsaved changes will be discarded. Are you sure you want to leave?`
                    )}
                />
                {this.state.answers.map((answer) => {
                    let key = (answer._id) ? answer._id : "new question key";
                    const editMode = this.state.editAnswer && (answer._id === this.state.answerId || answer._id === undefined);
                    return (
                        <AnswerBoxView editMode={editMode}
                                       user={this.props.user}
                                       onSubmit={() => this.onSubmit(answer)}
                                       answer={answer}
                                       onAnswerChange={this.onAnswerChange}
                                       key={key}
                                       onEditClick={() => this.onEditClick(answer)}
                                       resetAnswer={this.resetAnswers}
                        />
                    )
                })}
                {(this.props.user && !this.state.editAnswer) &&
                <div key="edit-answer-button" style={{float: 'right', marginTop: 5}}>
                    <Button fab color="primary" onClick={this.addAnswer}>
                        <AddIcon/>
                    </Button>
                </div>}

            </div>
        )
    }
}

interface  StateToProps {
    user: UserDto
    question: Question;
    answers: Answer[];
}

interface DispatchToProps {
    editAnswer: (answer: Answer) => void;
    addAnswer: (answer: Answer) => void;
    upVoteAnswer: (answer: Answer) => void;
    downVoteAnswer: (answer: Answer) => void;
}

const mapStateToProps = (state: AppStoreState): StateToProps => ({
    user: state.auth.user,
    question: state.questionPage.questionPage.question,
    answers: state.questionPage.questionPage.answers
});

const mapDispatchToProps = (dispatch): DispatchToProps => ({
    editAnswer: (answer: Answer) => dispatch(AnswerActions.updateAnswer(answer)),
    addAnswer: (answer: Answer) => dispatch(AnswerActions.createAnswer(answer)),
    upVoteAnswer: (answer: Answer) => dispatch(AnswerActions.createAnswer(answer)),
    downVoteAnswer: (answer: Answer) => dispatch(AnswerActions.createAnswer(answer))
});

export const AnswerBoxesView = connect<StateToProps, DispatchToProps, any>(
    mapStateToProps, mapDispatchToProps)(AnswerBoxesComponent);