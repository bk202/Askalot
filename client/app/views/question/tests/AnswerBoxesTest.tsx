import * as React from "react";
import * as renderer from 'react-test-renderer';
import {UserDto} from "../../../../../server/dtos/auth/UserDto";
import {UserTypeEnum} from "../../../../../server/enums/UserTypeEnum";
import {FrontEndQuestionModels} from "../../../models/QuestionModels";
import {Route, Router} from "react-router";
import {RouterController} from "../../../api.controllers/RouterController";
import {createMuiTheme, MuiThemeProvider} from "material-ui/styles";
import {CommentDto} from "../../../../../server/dtos/q&a/CommentDto";
import {AnswerBoxesComponent} from "../subcomponents/AnswerBoxesComponent";
import Question = FrontEndQuestionModels.Question;


jest.mock('draft-js/lib/generateRandomKey', () => () => '123');
jest.mock('../../../components/CustomEditor/EditorFactory');
const e = require("../../../components/CustomEditor/EditorFactory");
const mMock = jest.fn();
e.EditorFactory.mockImplementation(() => {
    return {
        createRichEditor: () => null
    }
});

let muiTheme = createMuiTheme();
let user: UserDto = {
    _id: "something",
    company: undefined,
    email: "sheldon@email.com",
    points: 0,
    role: UserTypeEnum.NORMAL,
    university: undefined,
    username: "sheldon",
    verified: true,
    country: undefined
};
const functionStub = () => {
};
let questionStub: Question = new Question();
questionStub.author = user;
let propsStub = {
    user: user,
    question: questionStub,
    answers: [],
    editAnswer: functionStub,
    addAnswer: functionStub,
    /*
    * user: UserDto
    question: Question;
    answers: Answer[];
}

interface DispatchToProps {
    editAnswer: (answer: Answer) => void;
    addAnswer: (answer: Answer) => void;*/
};
describe('answer boxes tests', function () {
    test('render question logged in', () => {
        const component = renderer.create(
            <MuiThemeProvider theme={muiTheme}>
                <Router history={RouterController.history}>
                    <Route render={(props: any) => {
                        return <AnswerBoxesComponent {...propsStub} {...props} />
                    }}/>
                </Router>
            </MuiThemeProvider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


    test('render question box not logged in', () => {
        let tempPropsStub = propsStub;
        tempPropsStub.user = undefined;
        const component = renderer.create(
            <MuiThemeProvider theme={muiTheme}>
                <Router history={RouterController.history}>
                    <Route render={(props: any) => {
                        return <AnswerBoxesComponent {...propsStub} {...props} />
                    }}/>
                </Router>
            </MuiThemeProvider>);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
