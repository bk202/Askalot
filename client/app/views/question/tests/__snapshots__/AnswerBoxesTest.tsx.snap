// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`answer boxes tests render question box not logged in 1`] = `<div />`;

exports[`answer boxes tests render question logged in 1`] = `
<div>
  <div
    style={
      Object {
        "float": "right",
        "marginTop": 5,
      }
    }
  >
    <button
      className="MuiButtonBase-root-15 MuiButton-root-1 MuiButton-raised-8 MuiButton-fab-14 MuiButton-raisedPrimary-10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-3"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-17"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-18"
      />
    </button>
  </div>
</div>
`;
